File: Berger.Data/MsfaEntity/KPI/UniverseReachAnalysis.cs
using Berger.Data.Common;
using Berger.Data.MsfaEntity.Users;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Berger.Data.MsfaEntity.KPI
{
    public class UniverseReachAnalysis : AuditableEntity<int>
    {
        public string BusinessArea { get; set; } // Plant, Depot
        public string Territory { get; set; }
        public string FiscalYear { get; set; }
        public int OutletNumber { get; set; }
        public int DirectCovered { get; set; }
        public int IndirectCovered { get; set; }
        public int DirectTarget { get; set; }
        public int IndirectTarget { get; set; }
        //public int DirectActual { get; set; }
        //public int IndirectActual { get; set; }
        public int IndirectManual { get; set; }
    }
}

File: BergerMsfaApi/ClientApp/src/app/Shared/Entity/KPI/UniverseReachAnalysis.ts
import { Status } from '../../Enums/status';
import { QueryObject } from '../Common/query-object';

export class UniverseReachAnalysis {
    id: number;
    businessArea: string;
    territory: string;
    fiscalYear: string;
    outletNumber: number;
    directCovered: number;
    indirectCovered: number;
    directTarget: number;
    indirectTarget: number;
    indirectManual: number;

    constructor(init?: Partial<UniverseReachAnalysis>) {
        Object.assign(this, init);
    }

    clear() {
        this.id = null;
    }
}

export class SaveUniverseReachAnalysis {
    id: number;
    businessArea: string;
    territory: string;
    outletNumber: number;
    directCovered: number;
    indirectCovered: number;
    directTarget: number;
    indirectTarget: number;
    indirectManual: number;
    
    constructor(init?: Partial<SaveUniverseReachAnalysis>) {
        Object.assign(this, init);
    }

    clear() {
        this.id = null;
    }
}

export class UniverseReachAnalysisQuery extends QueryObject {
    businessArea: string;
    territories: string[];

    constructor(init?: Partial<UniverseReachAnalysisQuery>) {
        super();
        Object.assign(this, init);
    }

    clear() {
    }
}
